src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/get_file_request.cc

```diff
@@ -0,0 +1,351 @@
+                           : NULL;
+
+    SendResponse(success, result, info, &msg_params);
+    return;
+  }
+  smart_objects::SmartObject response_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+  std::vector<uint8_t> bin_data;
+
+  if (event_message[strings::msg_params].keyExists(strings::file_type)) {
+    response_params[strings::file_type] =
+        event_message[strings::msg_params][strings::file_type];
+  }
+
+  if (event_message[strings::msg_params].keyExists(strings::file_path)) {
+    std::string full_path =
+        event_message[strings::msg_params][strings::file_path].asString();
+
+    if (!file_system::FileExists(full_path)) {
+      LOG4CXX_ERROR(logger_, "File " << full_path << " does not exist");
+      SendResponse(false,
+                   mobile_apis::Result::FILE_NOT_FOUND,
+                   "File does not exist",
+                   &response_params);
+      return;
+    }
+
+    const uint64_t file_size = file_system::FileSize(full_path);
+    if ((*message_)[strings::msg_params].keyExists(strings::length)) {
+      length_ = (*message_)[strings::msg_params][strings::length].asInt();
+    } else {
+      length_ = file_size;
+    }
+    if ((*message_)[strings::msg_params].keyExists(strings::offset)) {
+      offset_ = (*message_)[strings::msg_params][strings::offset].asInt();
+    }
+
+    if (offset_ > file_size) {
+      LOG4CXX_ERROR(logger_,
+                    "Offset " << offset_ << " greater than file size "
+                              << file_size);
+      SendResponse(false,
+                   mobile_apis::Result::INVALID_DATA,
+                   "Offset greater than file size",
+                   &response_params);
+      return;
+    }
+    if (length_ > file_size - offset_) {
+      LOG4CXX_ERROR(logger_,
+                    "Length " << length_ << " greater than file size - offset"
+                              << file_size);
+      SendResponse(false,
+                   mobile_apis::Result::INVALID_DATA,
+                   "Length greater than file size - offset",
+                   &response_params);
+      return;
+    }
+    if (!file_system::ReadBinaryFile(full_path, bin_data, offset_, length_)) {
+      LOG4CXX_ERROR(logger_, "Failed to read from file: " << full_path);
+      SendResponse(false,
+                   mobile_apis::Result::GENERIC_ERROR,
+                   "Failed to read from file",
+                   &response_params);
+      return;
+    }
+
+    const uint32_t crc_calculated = GetCrc32CheckSum(bin_data);
+    response_params[strings::crc32_check_sum] = crc_calculated;
+
+  } else {
+    LOG4CXX_ERROR(logger_, "HMI did not return a file path: ");
+    SendResponse(false,
+                 mobile_apis::Result::INVALID_DATA,
+                 "HMI did not return a file path",
+                 &response_params);
+    return;
+  }
+
+  SendResponse(true,
+               mobile_apis::Result::SUCCESS,
```

Comment:
Should pass `success` and `result` here, rather than just assuming SUCCESS

Reply:
```
if (!success) {
    auto msg_params = event_message[strings::msg_params];
    const char* info = msg_params.keyExists(strings::info)
                           ? msg_params[strings::info].asCharArray()
                           : NULL;

    SendResponse(success, result, info, &msg_params);
    return;
  }
```
would handle the success=false case. And for the result code I would have to check if it a valid result code(from the MOBILE_API) for the GetFile RPC. Like you mentioned if it sends a WARNING result code, that would be invalid for the GetFile RPC

Reply:
There's probably a way to check if the result code is valid for the RPC response but how should I handle a result code that does not exist in that list